app/src/main/java/ch/epfl/sdp/appart/database/FirestoreDatabaseService.java

```diff
@@ -240,19 +212,18 @@ public FirestoreDatabaseService() {
 
     }
 
-    private CompletableFuture<Boolean> updateUserDb(CompletableFuture<Boolean> res, User user){
-            db.document(user.getUserId())
-            .set(extractUserInfo(user))
-            .addOnCompleteListener(
-                task -> res.complete(task.isSuccessful()));
-            return res;
+    private CompletableFuture<Boolean> updateUserDb(CompletableFuture<Boolean> res, User user) {
+        db.document(user.getUserId())
+                .set(extractUserInfo(user))
+                .addOnCompleteListener(
+                        task -> res.complete(task.isSuccessful()));
```

Comment:
Add a failure case if the task is not successful.

Reply:
it's a task, the failure is given by `isSuccessful` returning false.

Reply:
oh I see what you mean, I should add a check on res to see if it has completed with false

Reply:
Yes, so that the caller can recover using exceptionally.

Reply:
In fact, that part of code belongs to a different pr that is still work in progress, so it's not up to me to deal with this.